src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/HttpClientBuilderExtensions.cs

```diff
@@ -335,16 +335,21 @@ public static IHttpClientBuilder ConfigureHttpMessageHandlerBuilder(this IHttpCl
         {
             ReserveClient(builder, typeof(TClient), builder.Name, validateSingleType);
 
-            builder.Services.AddTransient<TClient>(s =>
+            builder.Services.AddTransient<TClient>(s => AddTransientHelper<TClient>(s));
+
+            return builder;
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:RequiresUnreferencedCode",
+                Justification = "Workaround for https://github.com/mono/linker/issues/1416. Outer method has been annotated with DynamicallyAccessedMembers.")]
+            TClient1 AddTransientHelper<TClient1>(IServiceProvider s)
```

Comment:
Maybe `TClient1` => `TClientInner`?

Or better yet, maybe don't make the method nested.... Then you can just name it `TClient`.